chaosgcp/apphub/actions.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2024 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Dict, Any
+
+from google.cloud import compute_v1, apphub_v1
+from google.cloud import resourcemanager_v3
+
+from chaoslib.types import Configuration, Secrets
+from chaosgcp.lb.actions import (
+    inject_traffic_faults,
+    remove_fault_injection_traffic_policy,
+)
+
+__all__ = [
+    "inject_fault_if_url_map_exists_app_hub",
+    "remove_fault_if_url_map_exists_app_hub",
+]
+logger = logging.getLogger("chaostoolkit")
+
+
+def get_project_number(project_id: str):
+    """Retrieves the project number for the specified Google Cloud Project ID."""
+    client = resourcemanager_v3.ProjectsClient()
+    # Construct the project name (required format)
+    project_name = f"projects/{project_id}"
+    try:
+        project = client.get_project(name=project_name)
+        return project.name.split("/")[-1]
+        # Extract number from 'projects/123456789'
+    except Exception as e:
+        logger.error(f"Error retrieving project number: {e}")
```

Comment:
I might select to raise `ActivityFailed error here.

Reply:
ActivityFailed Is included .